DynamicConfigSplitBrain_whenConfigExistsInSmallerBrainOnlyTest.testSplitBrain random failure

Failed in PR builder: https://hazelcast-l337.ci.cloudbees.com/view/Lab/job/new-lab-fast-pr/14274/

Stacktrace:
```
java.lang.AssertionError: Cluster size is not correct. Expected: 2, actual: 1, instance index: 0
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:950)
	at com.hazelcast.test.HazelcastTestSupport$13.run(HazelcastTestSupport.java:972)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1145)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:968)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:962)
	at com.hazelcast.test.SplitBrainTestSupport.assertSplitBrainCreated(SplitBrainTestSupport.java:269)
	at com.hazelcast.test.SplitBrainTestSupport.createSplitBrain(SplitBrainTestSupport.java:258)
	at com.hazelcast.test.SplitBrainTestSupport.doIteration(SplitBrainTestSupport.java:191)
	at com.hazelcast.test.SplitBrainTestSupport.testSplitBrain(SplitBrainTestSupport.java:184)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
13:13:36,110  WARN |testSplitBrain| - [MockJoiner] hz._hzInstance_22_dev.cached.thread-4 - [127.0.0.1]:5022 [dev] [3.9.4-SNAPSHOT] Error during join check!
java.util.concurrent.ExecutionException: com.hazelcast.spi.exception.RetryableIOException: Packet not send to -> [127.0.0.1]:5023
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:171) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.AbstractJoiner.sendSplitBrainJoinMessage(AbstractJoiner.java:408) [classes/:?]
	at com.hazelcast.test.mocknetwork.MockJoiner.searchForOtherClusters(MockJoiner.java:156) [test-classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.searchForOtherClusters(SplitBrainHandler.java:75) [classes/:?]
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.run(SplitBrainHandler.java:42) [classes/:?]
	at com.hazelcast.spi.impl.executionservice.impl.DelegateAndSkipOnConcurrentExecutionDecorator$DelegateDecorator.run(DelegateAndSkipOnConcurrentExecutionDecorator.java:66) [classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_161]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_161]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_161]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [classes/:?]
Caused by: com.hazelcast.spi.exception.RetryableIOException: Packet not send to -> [127.0.0.1]:5023
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeRemote(Invocation.java:575) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:552) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:513) ~[classes/:?]
```

There are even more scary exceptions on a successful (!) local execution of this test:
```java
16:53:33,177 ERROR |testSplitBrain| - [MigrationManager] hz._hzInstance_2_dev.migration - [127.0.0.1]:5002 [dev] [3.9.4-SNAPSHOT] Promotion commit to [127.0.0.1]:5002 failed for [MigrationInfo{uuid=24e83bad-e02f-4bf3-a0ef-bdc9134d14e3, partitionId=2, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=d41927b1-3e6e-4d46-aa0e-8deecd1610a8, partitionId=14, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=f769b696-ea05-4e68-bd0a-0752ae51e650, partitionId=21, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=e60a1515-1e07-4749-9685-a843799f5c87, partitionId=26, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=dfd77fd4-b1b1-40fa-bd1e-1b062760c67e, partitionId=31, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=c5f2573d-e314-4970-94ef-f318344044e1, partitionId=34, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=c6a42b66-fbb8-4b69-801c-41abcfafcac5, partitionId=46, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=05c577eb-f061-42ad-ba74-8e0bbf2a67da, partitionId=47, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=c84b0291-cac5-44b8-b4ac-3b28589fbc23, partitionId=48, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=238b4eaf-8047-4638-9cc7-0dfbc3bc39dc, partitionId=52, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=f19ac466-8869-4eb3-8520-92da35256bed, partitionId=61, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=6437baf8-a576-41c0-b105-d22a9ef86aad, partitionId=62, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=ebd4f536-341f-4930-8a8c-edef1967a29b, partitionId=65, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=4e1ef5db-845a-4b8e-8bff-d5adfee78e40, partitionId=75, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=e5308be7-0fc7-4c7b-8cdf-03e3785582b2, partitionId=86, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=d33159bd-405b-4d2c-9cf4-7a6f5f0b3830, partitionId=99, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=21c5cba9-d216-4032-8367-3b68343a91d7, partitionId=110, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=0d2d3db0-98fd-4985-bf23-c1f80a14eed6, partitionId=118, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=b48a4524-a314-4268-9cca-a5f53ac6634a, partitionId=122, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=c9add3d1-2c00-48df-9386-c1342b52ab0f, partitionId=136, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=d13ddc73-4fb3-411e-95a7-2e38d00fdff8, partitionId=144, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=c929ae5a-e8c4-4fd8-ad4a-0fdda243e5d2, partitionId=151, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=adb91b15-a343-44f6-8edc-e80ef2df0655, partitionId=174, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=9f2a7dc6-889f-44e1-b28a-a1b04f012159, partitionId=188, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=b2da46ac-c7f4-4b32-8bbc-9c72544809ae, partitionId=191, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=0d98bb7c-61c0-4e0d-9422-41ce5cbde62f, partitionId=192, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=30f262c5-4a7c-49d3-8d68-b726fd252a2e, partitionId=198, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=7bc25bef-c12e-46f7-8aed-67c2561e2c41, partitionId=199, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=8dbce5c1-769e-4fcb-900e-731c3f7e50ee, partitionId=201, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=33838546-9a1c-4388-bce0-2d199492c80f, partitionId=206, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=d4247fc1-9293-4944-9878-b12ced8b5bc5, partitionId=209, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=fed7d85d-65c9-4fbe-9052-982978ae72eb, partitionId=210, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=179f22eb-84f9-4483-8bec-1e6fa4569d4f, partitionId=212, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=beaf2306-6ad1-4b79-9cf1-ddbdede310e5, partitionId=214, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=701a5080-8b31-47c4-85fc-7b942ab9adf8, partitionId=223, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=ffdfa725-8e63-4a6d-a281-f7128ac184be, partitionId=236, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=2367cc46-6762-4e18-b7e4-1037e62a2874, partitionId=241, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=fe0cb0da-fd25-4f27-b650-00a17a595f31, partitionId=254, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=9b6449f2-3c37-4706-8737-b20a9cea6649, partitionId=261, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}, MigrationInfo{uuid=546f6c46-8551-4154-983a-a43dd7f24c6f, partitionId=264, source=null, sourceUuid=null, sourceCurrentReplicaIndex=-1, sourceNewReplicaIndex=-1, destination=[127.0.0.1]:5002, destinationUuid=de3f1a7d-cd14-40d4-a55a-e1b73dacfcab, destinationCurrentReplicaIndex=1, destinationNewReplicaIndex=0, master=[127.0.0.1]:5002, processing=false, status=SUCCESS}]
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Caller is not master node! Caller: [127.0.0.1]:5002, Master: null
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:147) ~[classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.commitPromotionsToDestination(MigrationManager.java:1317) [classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.commitPromotionMigrations(MigrationManager.java:1200) [classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.promoteBackupsForMissingOwners(MigrationManager.java:1187) [classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.run(MigrationManager.java:1128) [classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationThread.processTask(MigrationThread.java:121) [classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationThread.doRun(MigrationThread.java:98) [classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationThread.run(MigrationThread.java:67) [classes/:?]
Caused by: java.lang.IllegalStateException: Caller is not master node! Caller: [127.0.0.1]:5002, Master: null
	at com.hazelcast.internal.partition.operation.PromotionCommitOperation.beforeRun(PromotionCommitOperation.java:92) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:188) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:569) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:554) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:513) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:207) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:60) ~[classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.commitPromotionsToDestination(MigrationManager.java:1315) ~[classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.commitPromotionMigrations(MigrationManager.java:1200) ~[classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.promoteBackupsForMissingOwners(MigrationManager.java:1187) ~[classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationManager$RepairPartitionTableTask.run(MigrationManager.java:1128) [classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationThread.processTask(MigrationThread.java:121) ~[classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationThread.doRun(MigrationThread.java:98) ~[classes/:?]
	at com.hazelcast.internal.partition.impl.MigrationThread.run(MigrationThread.java:67) ~[classes/:?]
	at ------ submitted from ------.(Unknown Source) ~[?:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:127) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79) ~[classes/:?]
	... 8 more
```

New failure in http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/5459/
**Error Message**
```
expected:<OBJECT> but was:<BINARY>
```
**Stacktrace**
```
java.lang.AssertionError: expected:<OBJECT> but was:<BINARY>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.internal.dynamicconfig.DynamicConfigSplitBrain_whenConfigExistsInSmallerBrainOnlyTest.onAfterSplitBrainHealed(DynamicConfigSplitBrain_whenConfigExistsInSmallerBrainOnlyTest.java:52)
	at com.hazelcast.test.SplitBrainTestSupport.doIteration(SplitBrainTestSupport.java:227)
	at com.hazelcast.test.SplitBrainTestSupport.testSplitBrain(SplitBrainTestSupport.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```
**Notable exception**
```
18:33:36,382 ERROR |testSplitBrain| - [ClusterMergeTask] hz.dreamy_mayer.cached.thread-4 - [127.0.0.1]:5703 [dev] [4.0-SNAPSHOT] While merging...
com.hazelcast.core.HazelcastException: Error while merging configurations
	at com.hazelcast.internal.dynamicconfig.ClusterWideConfigurationService$Merger.run(ClusterWideConfigurationService.java:523) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_181]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_181]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [classes/:?]
Caused by: java.lang.IllegalStateException: Attempt to reuse the same operation in multiple invocations. Operation is com.hazelcast.internal.dynamicconfig.DynamicConfigPreJoinOperation{serviceName='configuration-service', identityHash=1000746862, partitionId=-1, replicaIndex=0, callId=10, invocationTime=1578508416378 (2020-01-08 18:33:36.378), waitTimeout=-1, callTimeout=60000}
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:521) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:236) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.invokeOnTarget(OperationServiceImpl.java:337) ~[classes/:?]
	at com.hazelcast.internal.util.InvocationUtil$InvokeOnMemberFunction.invokeOnMember(InvocationUtil.java:151) ~[classes/:?]
	at com.hazelcast.internal.util.InvocationUtil$InvokeOnMemberFunction.apply(InvocationUtil.java:128) ~[classes/:?]
	at com.hazelcast.internal.util.InvocationUtil$InvokeOnMemberFunction.apply(InvocationUtil.java:108) ~[classes/:?]
	at com.hazelcast.internal.util.IterableUtil$1.next(IterableUtil.java:54) ~[classes/:?]
	at com.hazelcast.internal.util.futures.ChainingFuture.advanceOrComplete(ChainingFuture.java:69) ~[classes/:?]
	at com.hazelcast.internal.util.futures.ChainingFuture.lambda$registerCallback$0(ChainingFuture.java:51) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture$WhenCompleteNode.lambda$execute$0(AbstractInvocationFuture.java:1527) ~[classes/:?]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402) ~[?:1.8.0_181]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[?:1.8.0_181]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[?:1.8.0_181]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[?:1.8.0_181]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[?:1.8.0_181]
18:33:36,395  INFO |testSplitBrain| - [LifecycleService] hz.dreamy_mayer.cached.thread-4 - [127.0.0.1]:5703 [dev] [4.0-SNAPSHOT] [127.0.0.1]:5703 is MERGED
```

--- a/hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java
@@ -501,23 +501,24 @@ public class ClusterWideConfigurationService implements PreJoinAwareService,
         if (noConfigurationExist(allConfigurations)) {
             return null;
         }
-        return new Merger(nodeEngine, new DynamicConfigPreJoinOperation(allConfigurations, ConfigCheckMode.SILENT));
+        return new Merger(nodeEngine, allConfigurations);
     }
 
     public static class Merger implements Runnable {
         private final NodeEngine nodeEngine;
-        private final Operation replicationOperation;
+        private final IdentifiedDataSerializable[] allConfigurations;
 
-        public Merger(NodeEngine nodeEngine, Operation replicationOperation) {
+        public Merger(NodeEngine nodeEngine, IdentifiedDataSerializable[] allConfigurations) {
             this.nodeEngine = nodeEngine;
-            this.replicationOperation = replicationOperation;
+            this.allConfigurations = allConfigurations;
         }
 
         @Override
         public void run() {
             try {
                 Future<Object> future = invokeOnStableClusterSerial(nodeEngine,
-                        () -> replicationOperation, CONFIG_PUBLISH_MAX_ATTEMPT_COUNT);
+                        () -> new DynamicConfigPreJoinOperation(allConfigurations, ConfigCheckMode.SILENT),
+                        CONFIG_PUBLISH_MAX_ATTEMPT_COUNT);
                 waitForever(singleton(future), FutureUtil.RETHROW_EVERYTHING);
             } catch (Exception e) {
                 throw new HazelcastException("Error while merging configurations", e);

--- a/hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java
@@ -39,6 +39,7 @@ import com.hazelcast.config.TopicConfig;
 import com.hazelcast.core.HazelcastException;
 import com.hazelcast.internal.cluster.ClusterService;
 import com.hazelcast.internal.cluster.ClusterVersionListener;
+import com.hazelcast.internal.serialization.Data;
 import com.hazelcast.internal.serialization.SerializationService;
 import com.hazelcast.internal.services.CoreService;
 import com.hazelcast.internal.services.ManagedService;
@@ -46,12 +47,12 @@ import com.hazelcast.internal.services.PreJoinAwareService;
 import com.hazelcast.internal.services.SplitBrainHandlerService;
 import com.hazelcast.internal.util.FutureUtil;
 import com.hazelcast.logging.ILogger;
-import com.hazelcast.internal.serialization.Data;
 import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
 import com.hazelcast.spi.impl.InternalCompletableFuture;
 import com.hazelcast.spi.impl.NodeEngine;
 import com.hazelcast.spi.impl.operationservice.Operation;
 import com.hazelcast.version.Version;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 import java.util.ArrayList;
 import java.util.Collection;
@@ -167,7 +168,7 @@ public class ClusterWideConfigurationService implements PreJoinAwareService,
     }
 
     private IdentifiedDataSerializable[] collectAllDynamicConfigs() {
-        List<IdentifiedDataSerializable> all = new ArrayList<IdentifiedDataSerializable>();
+        List<IdentifiedDataSerializable> all = new ArrayList<>();
         for (Map<?, ? extends IdentifiedDataSerializable> entry : allConfigurations) {
             Collection<? extends IdentifiedDataSerializable> values = entry.values();
             all.addAll(values);
@@ -508,6 +509,7 @@ public class ClusterWideConfigurationService implements PreJoinAwareService,
         private final NodeEngine nodeEngine;
         private final IdentifiedDataSerializable[] allConfigurations;
 
+        @SuppressFBWarnings(value = "EI_EXPOSE_REP2")
         public Merger(NodeEngine nodeEngine, IdentifiedDataSerializable[] allConfigurations) {
             this.nodeEngine = nodeEngine;
             this.allConfigurations = allConfigurations;

